app/models/student_loans/school_eligibility.rb

```diff
@@ -58,7 +58,11 @@ def eligible_phase?
     end
 
     def eligible_special_school?
-      @school.phase == "not_applicable" && @school.special? && @school.school_type != "special_post_16_institutions"
+      @school.phase == "not_applicable" && @school.has_statutory_high_age_over_eleven? && @school.special? && @school.school_type != "special_post_16_institutions"
+    end
+
+    def eligible_alternative_provision_school?
+      @school.phase == "not_applicable" && @school.has_statutory_high_age_over_eleven? && @school.alternative_provision?
```

Comment:
There's a part of me that feels like the combination of these particular checks define a specific "type" of school. If we could come up with a name for that, then we could move this logic onto the `School` model and better communicate what this represents in terms of the characteristics of the school, rather than "these things make it an eligible alternative provision school". Same applies to the `eligible_special_school?` method. Does that make sense?